File: totosuki/ABC/C/274.py
N = int(input())
A = list(map(int, input().split()))
d = {1:0}

for i in range(1, N+1):
    n1 = 2*i
    n2 = 2*i+1
    d[n1] = d[A[i-1]] + 1
    d[n2] = d[A[i-1]] + 1

ans = [d[i] for i in range(1, 2*N+2)]
print(*ans, sep = "\n")

File: OCH_Tao/ABC/B/274.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [, W] = input.shift().split(" ").map(x => parseInt(x, 10));
let X = new Array(W).fill(0);
for (const C of input) {
  for (let i = 0; i < W; i++) {
    if (C[i] === "#") {
      X[i]++;
    }
  }
}
console.log(X.join(" "));